pkg/aws/eks/eks.go

```diff
@@ -0,0 +1,360 @@
+package eks
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/pricing"
+	"github.com/aws/aws-sdk-go-v2/service/pricing/types"
+	"github.com/prometheus/client_golang/prometheus"
+
+	cloudcostexporter "github.com/grafana/cloudcost-exporter"
+	ec2client "github.com/grafana/cloudcost-exporter/pkg/aws/services/ec2"
+	pricingClient "github.com/grafana/cloudcost-exporter/pkg/aws/services/pricing"
+	"github.com/grafana/cloudcost-exporter/pkg/provider"
+)
+
+const (
+	subsystem = "eks"
+)
+
+// These were generated by analysing Grafana Labs spend in GCP and finding the ratio of CPU to Memory spend by instance type.
+// It's an imperfect approximation, but it's better than nothing.
+var cpuToCostRation = map[string]float64{
+	"Compute optimized": 0.88,
+	"Memory optimized":  0.48,
+	"General purpose":   0.65,
+}
+var clusterTags = []string{"cluster", "eks:cluster-name", "aws:eks:cluster-name"}
+
+var (
+	ErrRegionNotFound       = errors.New("no region found")
+	ErrInstanceTypeNotFound = errors.New("no instance type found")
+	ErrClientNotFound       = errors.New("no client found")
+	ListSpotPricesErr       = errors.New("error listing spot prices")
+	GeneratePricingMapErr   = errors.New("error generating pricing map")
+)
+
+var (
+	InstanceCPUHourlyCostDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(cloudcostexporter.MetricPrefix, subsystem, "instance_cpu_usd_per_core_hour"),
+		"The cpu cost a compute instance in USD/(core*h)",
+		[]string{"instance", "region", "family", "machine_type", "cluster", "price_tier", "provider"},
```

Comment:
is there a point to have provider here? ... eks is already in the name of the metric

Reply:
I suppose not. It doesn't make sense since we can't aggregate metrics between providers anyway.